src/app/utils/helpers.tsx

```diff
@@ -5,15 +5,35 @@ import {
   TBundle,
   IResource,
   TSupportedResource,
+  EResource,
 } from '@/types/fhir.types';
 
 export const extractResource = <TResource extends keyof IResource>(
   bundle: TBundle,
   resourceType: TResource,
-) =>
-  bundle.entry
-    ?.filter(({ resource }) => resource.resourceType === resourceType)
-    .map(({ resource }) => resource) as TType<TResource>[];
+) => {
+  const filterResourceByType = (type: keyof IResource) =>
+    bundle.entry
+      ?.filter(({ resource }) => resource.resourceType === type)
+      .map(({ resource }) => resource) as TType<TResource>[];
+
+  // Define the dependent resources based on the provided resourceType
+  const dependentResources: Record<string, (keyof IResource)[]> = {
+    Medication: [EResource.MedicationStatement],
+    Composition: [EResource.Patient, EResource.Condition],
+  };
```

Comment:
Question: just confirming, the intended purposes of this definition as that the resource defined as a key, should also resolve the other resources defined within the array?

For any resource not defined in here, it means they have no dependant resources to extract and include. And only their own resource will be returned (IE, Encounter isnt defined in here, so for Encounter, it contains only Encounter resource data)

Reply:
Hello @BMartinos,

Yes, exactly! My intention was to gather in one place only the resources that require information from external sources. Each key represents a main resource, while its values represent a list of related resources that it will need. I then used this dependentResources object to differentiate how we will pass data to each component.

For example:

- If we're dealing with the `Encounter` component, which does not belong to dependentResources, we will only pass the Encounter data to the component.

-  On the other hand if we're working with the `Medication` component, we will pass both the Medication resource data and the MedicationStatement resource data.

Please note that this is not the definitive structure for `dependentResources`; it’s just a preliminary one to test the solution that came to my mind.